test/models/vrp_consistency_test.rb

```diff
@@ -311,5 +311,26 @@ def test_dates_cannot_be_mixed_with_indices
         Models::Vrp.filter(vrp)
       end
     end
+
+    def test_switched_lapses_are_rejected
+      vrp = VRP.scheduling
+      vrp[:services].first[:minimum_lapse] = 7
+      vrp[:services].first[:maximum_lapse] = 14
+      vrp[:configuration][:preprocessing][:partitions] = TestHelper.vehicle_and_days_partitions
+      Models::Vrp.check_consistency(vrp) # no raise
+
+      vrp[:services].first[:visits_number] = 3
+      vrp[:services].first[:minimum_lapse] = 14
+      vrp[:services].first[:maximum_lapse] = 7
+      raised = false
+      begin
+        Models::Vrp.check_consistency(vrp)
+      rescue OptimizerWrapper::DiscordantProblemError => e
+        raised = true
+        assert_match 'Maximum lapse should always be bigger than minimum lapse', e.message
+      end
+
+      assert raised, 'This VRP is not consistent, it should raise an error.'
```

Comment:
```suggestion
      error = assert_raises DiscordantProblemError do
        Models::Vrp.check_consistency(vrp)
      end
      assert_equal 'Maximum lapse should always be bigger than minimum lapse', error.message
```
assert_raises returns the error
assert_match is a regexp match and because if that its fail message is more complex

For example 
`assert_equal 'Centroid 0 is initialised with an incompatible service -- 1', error.message` returns
```
 FAIL["test_skills_taken_into_account", #<Minitest::Reporters::Suite:0x000055e8b06549f8 @name="SplitClusteringTest">, 1.7672128429403529]
 test_skills_taken_into_account#SplitClusteringTest (1.77s)
        --- expected
        +++ actual
        @@ -1 +1 @@
        -"Centroid 0 is initialised with an incompatible service -- 1"
        +"Centroid 0 is initialised with an incompatible service -- 0"
        /home/halil/dev/optimizer-api/test/lib/interpreters/split_clustering_test.rb:345:in `test_skills_taken_into_account'
```
while it returns the following with `assert_match`
```
 FAIL["test_skills_taken_into_account", #<Minitest::Reporters::Suite:0x0000562af0b90f98 @name="SplitClusteringTest">, 1.6732092190068215]
 test_skills_taken_into_account#SplitClusteringTest (1.67s)
        Expected /Centroid\ 0\ is\ initialised\ with\ an\ incompatible\ service\ \-\-\ 1/ to match "Centroid 0 is initialised with an incompatible service -- 0".
        /home/halil/dev/optimizer-api/test/lib/interpreters/split_clustering_test.rb:345:in `test_skills_taken_into_account'
```

Reply:
I think it is better to accept the suggestions instead of copy/pasting because it preservers code ownership. Unless the suggestion needs modification or if it is not a working code...